core/src/main/java/de/jplag/merging/MatchMerging.java

```diff
@@ -43,29 +46,31 @@ public MatchMerging(JPlagOptions options) {
      */
     public JPlagResult mergeMatchesOf(JPlagResult result) {
         long timeBeforeStartInMillis = System.currentTimeMillis();
-
         List<JPlagComparison> comparisons = new ArrayList<>(result.getAllComparisons());
-        List<JPlagComparison> comparisonsMerged = new ArrayList<>();
-
-        ProgressBarLogger.iterate(ProgressBarType.MATCH_MERGING, comparisons, comparison -> {
-            numberOfMerges = 0;
-            Submission leftSubmission = comparison.firstSubmission().copy();
-            Submission rightSubmission = comparison.secondSubmission().copy();
-            List<Match> globalMatches = new ArrayList<>(comparison.matches());
-            globalMatches.addAll(comparison.ignoredMatches());
-            globalMatches = mergeNeighbors(globalMatches, leftSubmission, rightSubmission);
-            globalMatches = globalMatches.stream().filter(it -> it.length() >= options.minimumTokenMatch()).toList();
-            if (numberOfMerges >= options.mergingOptions().minimumRequiredMerges()) {
-                comparisonsMerged.add(new JPlagComparison(leftSubmission, rightSubmission, globalMatches, new ArrayList<>()));
-            } else {
-                comparisonsMerged.add(comparison);
-            }
-        });
+
+        ProgressBar progressBar = ProgressBarLogger.createProgressBar(ProgressBarType.MATCH_MERGING, comparisons.size());
+        List<JPlagComparison> comparisonsMerged = comparisons.parallelStream().map(it -> mergeMatchesOf(it, progressBar)).collect(toList());
```

Comment:
```suggestion
        List<JPlagComparison> comparisonsMerged = comparisons.parallelStream().map(it -> mergeMatchesOf(it, progressBar)).toList();
```

Reply:
Interesting that sonar didn't catch this...

Reply:
Oh wow [it did](https://sonarcloud.io/project/issues?issueStatuses=OPEN%2CCONFIRMED&sinceLeakPeriod=true&pullRequest=2234&id=jplag_JPlag), but it was not interested in telling us! :D